src/main/java/com/cleanroommc/groovyscript/compat/mods/woot/Spawning.java

```diff
@@ -0,0 +1,159 @@
+package com.cleanroommc.groovyscript.compat.mods.woot;
+
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.compat.mods.ModSupport;
+import com.cleanroommc.groovyscript.core.mixin.woot.SpawnRecipeRepositoryAccessor;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import ipsis.Woot;
+import ipsis.woot.farming.ISpawnRecipe;
+import ipsis.woot.farming.SpawnRecipe;
+import ipsis.woot.util.WootMobName;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fml.common.registry.EntityEntry;
+import org.apache.commons.lang3.tuple.Pair;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+
+public class Spawning extends VirtualizedRegistry<Pair<WootMobName, SpawnRecipe>> {
+
+    public Spawning() {
+        super();
+    }
+
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    @Override
+    public void onReload() {
+        restoreFromBackup().forEach(pair -> ((SpawnRecipeRepositoryAccessor) Woot.spawnRecipeRepository).getRecipes().put(pair.getKey(), pair.getValue()));
+        removeScripted().forEach(pair -> ((SpawnRecipeRepositoryAccessor) Woot.spawnRecipeRepository).getRecipes().remove(pair.getKey()));
+        //((SpawnRecipeRepositoryAccessor) Woot.spawnRecipeRepository).getDefaultSpawnRecipe().setEfficiency(true);
+        ((SpawnRecipeRepositoryAccessor) Woot.spawnRecipeRepository).getDefaultSpawnRecipe().getItems().clear();
+        ((SpawnRecipeRepositoryAccessor) Woot.spawnRecipeRepository).getDefaultSpawnRecipe().getFluids().clear();
+    }
+
+    public void addDefaultItem(ItemStack item) {
+        ((SpawnRecipeRepositoryAccessor) Woot.spawnRecipeRepository).getDefaultSpawnRecipe().addIngredient(item);
+    }
+
+    public void addDefaultFluid(FluidStack fluid) {
+        ((SpawnRecipeRepositoryAccessor) Woot.spawnRecipeRepository).getDefaultSpawnRecipe().addIngredient(fluid);
+    }
+
+    //public void setDefaultEfficiency(boolean efficiency) {
+    //    ((SpawnRecipeRepositoryAccessor) Woot.spawnRecipeRepository).getDefaultSpawnRecipe().setEfficiency(efficiency);
+    //}
+
```

Comment:
Whats the deal with this?

Reply:
as far as i can tell, efficiency doesnt do anything. however, i could be missing it somehow, or it could be changed to do something, and so this code is merely commented out, able to be uncommented and compat added if it ever would be useful.